src/error.rs

```diff
@@ -0,0 +1,61 @@
+use std::error;
+use std::fmt;
+/// The error type returned by methods in this crate.
+pub enum Error<E> {
+    /// Manager Errors
+    Inner(E),
+    /// Timeout
+    Timeout,
+    /// BadConn
+    BadConn,
+    /// The pool has been closed or dropped
+    PoolClosed,
+}
+
+impl<E> From<E> for Error<E> {
+    fn from(e: E) -> Error<E> {
+        Error::Inner(e)
+    }
+}
+
+impl<E> fmt::Display for Error<E>
+where
+    E: fmt::Display + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Inner(ref err) => write!(f, "{}", err),
+            Error::Timeout => write!(f, "Timed out in mobc"),
```

Comment:
Maybe `Timed out in the connection pool`. Our users don't need to know the name of the library.